### PR TITLE
Converted R&D to use networks.

### PR DESCRIPTION
**Pull request by biscuitWizard**

### CODE REVIEW

**Review by MistakeNot4892 — DISMISSED** [review]

**Comment by biscuitWizard** [comment]
image

File: code/modules/research/design_database_analyzer.dm [hunk 1]
```diff
@@ -71,8 +69,17 @@
 				cached_materials -= mat
 		return TRUE
 
+	if(!(stat & (BROKEN|NOPOWER)))
+		interface_interact(user)
+		return TRUE
```
Comment (afterthought2) [inline code/modules/research/design_database_analyzer.dm]: This shouldn't be called from here.

Comment (biscuitWizard) [reply code/modules/research/design_database_analyzer.dm]: ???

Comment (biscuitWizard) [reply code/modules/research/design_database_analyzer.dm]: Why can't I open the network_device UI from attack_hand?

Comment (afterthought2) [reply code/modules/research/design_database_analyzer.dm]: Because it's redundant with `interface_interact` below, except that it runs the wrong checks.

Comment (biscuitWizard) [reply code/modules/research/design_database_analyzer.dm]: weird, but that's the only way it works?

Comment (biscuitWizard) [reply code/modules/research/design_database_analyzer.dm]: Testing, give me a minute.

Comment (biscuitWizard) [reply code/modules/research/design_database_analyzer.dm]: you're right, it works. Pushed.

**Comment by comma** [comment]
No way to change network settings for fabricator, design DB and design console?

**Comment by biscuitWizard** [comment]
> No way to change network settings for fabricator, design DB and design console?

Nope.

If that's unacceptable, mark PR as DNM. I'll get around to it when I finish the major feature depending on this code downstream.

**Comment by biscuitWizard** [comment]
can remove DNM. Added network UI to all the machines that need it.

**Review by MistakeNot4892 — APPROVED** [review]

**Pull request merged**